assets/js/autofill.js

```diff
@@ -52,169 +52,220 @@ jQuery(document).ready(function( $ ) {
-            jQuery('#billing_address_1').val(place.address_components[i]['long_name']);
-          }
-        }
-      }
-      // adding data to street address field
-      if(addressType == 'route') {
-        var addr = jQuery('#billing_address_1').val();
-        if(addr != ''){
-          addr = addr +' '+ place.address_components[i]['long_name'];
-          jQuery('#billing_address_1').val(addr);
-        } else {
-          jQuery('#billing_address_1').val(place.address_components[i]['long_name']);
-        }
-      }
 
-      // filling state field
-      if(addressType == 'administrative_area_level_1'){
-        var state = place.address_components[i]['short_name'];
-        setTimeout(function(){
-          jQuery('#billing_state').val(state);
-          jQuery('#billing_state').trigger('change');
-        },1500);
-      }
-
-      // filling second address field
-      if(addressType == 'neighborhood'){
-        jQuery('#billing_address_2').val(place.address_components[i]['long_name']);
-      } else if(addressType == 'sublocality_level_3'){
-        jQuery('#billing_address_2').val(place.address_components[i]['long_name']);
-      } else if(addressType == 'sublocality_level_2'){
-        jQuery('#billing_address_2').val(place.address_components[i]['long_name']);
-      }
-
-      // filling location
-      if(addressType == 'locality'){
-        jQuery('#billing_city').val(place.address_components[i]['long_name']);
-      }
-      // filling postal code
-      if(addressType == 'postal_code'){
-        jQuery('#billing_postcode').val(place.address_components[i]['long_name']);
-      }
-
-    }
-
-    if( wcaf.enable_billing_phone ) {
-      if (place.hasOwnProperty("international_phone_number") && place.international_phone_number){
+			// Components we use 'short_name'.
+			if ( components_short_name.includes(addressType) ) {
+				address_data[addressType] = place.address_components[i]['short_name'];
+			}
+
+			// Components we use 'long_name'.
+			if ( components_long_name.includes(addressType) ) {
+				address_data[addressType] = place.address_components[i]['long_name'];
+			}
+		}
+		// If not filled in type 0, may try with the type 1.
+    for (var i = 0; i < place.address_components.length; i++) {
+			try {
+				// May not exists ...types[1]
+				var addressType = place.address_components[i].types[1];
+			} catch(err) {
+				continue;
+			}
+			if (address_data[addressType] !== '') {
+				// Case already have data for this addressType.
+				continue;
+			}
+
+			// Components we use 'short_name'.
+			if ( components_short_name.includes(addressType) ) {
+				address_data[addressType] = place.address_components[i]['short_name'];
+			}
+
+			// Components we use 'long_name'.
+			if ( components_long_name.includes(addressType) ) {
+				address_data[addressType] = place.address_components[i]['long_name'];
+			}
+		}
+
+		// filling state field after the country is set and field updated.
```

Comment:
Sometimes there may not be any state field. For example, Afghanistan does not have one.